src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -211,152 +207,78 @@ public bool TryGetTelemetryId(out Guid telemetryId)
-                    // a medium pri refactoring.
-                    //
-                    // Note: we do not do this for *high* pri refactorings (like 'rename').  These
-                    // are still very important and need to stay at the top (though still after high
-                    // pri fixes).
-                    var highPriRefactorings = refactorings.WhereAsArray(s => s.Priority == SuggestedActionSetPriority.High);
-                    var nonHighPriRefactorings = refactorings.WhereAsArray(s => s.Priority != SuggestedActionSetPriority.High)
-                                                             .SelectAsArray(s => WithPriority(s, SuggestedActionSetPriority.Low));
-
-                    var highPriFixes = fixes.WhereAsArray(s => s.Priority == SuggestedActionSetPriority.High);
-                    var nonHighPriFixes = fixes.WhereAsArray(s => s.Priority != SuggestedActionSetPriority.High);
-
-                    return highPriFixes.Concat(highPriRefactorings).Concat(nonHighPriFixes).Concat(nonHighPriRefactorings);
-                }
-            }
-
-            private static SuggestedActionSet WithPriority(SuggestedActionSet set, SuggestedActionSetPriority priority)
-                => new SuggestedActionSet(set.CategoryName, set.Actions, set.Title, priority, set.ApplicableToSpan);
-
-            private static ImmutableArray<SuggestedActionSet> OrderActionSets(
-                ImmutableArray<SuggestedActionSet> actionSets, TextSpan? selectionOpt)
-            {
-                return actionSets.OrderByDescending(s => s.Priority)
-                                 .ThenBy(s => s, new SuggestedActionSetComparer(selectionOpt))
-                                 .ToImmutableArray();
-            }
-
-            private static ImmutableArray<SuggestedActionSet> FilterActionSetsByTitle(ImmutableArray<SuggestedActionSet> allActionSets)
-            {
-                using var resultDisposer = ArrayBuilder<SuggestedActionSet>.GetInstance(out var result);
-                var seenTitles = new HashSet<string>();
-
-                foreach (var set in allActionSets)
-                {
-                    var filteredSet = FilterActionSetByTitle(set, seenTitles);
-                    if (filteredSet != null)
-                    {
-                        result.Add(filteredSet);
-                    }
-                }
-
-                return result.ToImmutable();
-            }
-
-            private static SuggestedActionSet? FilterActionSetByTitle(SuggestedActionSet set, HashSet<string> seenTitles)
+            private SuggestedActionSet? ConvertToSuggestedActionSet(UnifiedSuggestedActionSet? unifiedSuggestedActionSet)
             {
-                using var actionsDisposer = ArrayBuilder<ISuggestedAction>.GetInstance(out var actions);
-
-                foreach (var action in set.Actions)
+                if (unifiedSuggestedActionSet == null)
```

Comment:
when do we get null passed in?

Reply:
CodeFixSuggestedActions's FixAllFlavors may be null (even in the original implementation), so we may have null passed in for that scenario.